framework/platform/bpl/uci/cfg/bpl_cfg_uci.cpp

```diff
@@ -162,6 +162,132 @@ int cfg_uci_get_wireless(enum paramType type, int index, const char param[], cha
+        {
+            struct uci_option *o = uci_to_option(n);
+
+            if (o->type != UCI_TYPE_STRING)
+                continue;
+
+            if (strncmp(n->name, "ifname", MAX_UCI_BUF_LEN))
+                continue;
+
+            if (strncmp(interface_name, o->v.string, MAX_UCI_BUF_LEN))
+                continue;
+
+            // We reached the section containing the requested ifname
+            is_section_found = true;
+            break;
+        }
+
+        if (is_section_found) {
+            break;
+        }
+    }
+
+    if (type == TYPE_RADIO) {
+        //get radio name from .device param value
+        char deviceName[MAX_UCI_BUF_LEN] = {0};
+        uci_foreach_element(&s->options, n)
+        {
+            struct uci_option *o = uci_to_option(n);
+
+            if (strncmp(n->name, "device", MAX_UCI_BUF_LEN) == 0 && o->type == UCI_TYPE_STRING) {
+                strncpy_s(deviceName, MAX_UCI_BUF_LEN, o->v.string, MAX_UCI_BUF_LEN - 1);
+                continue;
+            }
+        }
+        uci_free_context(ctx);
+
+        if (deviceName[0] == '\0') {
+            // radio not found
+            ERROR("%s deviceName is mandatory param and not configured in %s\n", __func__,
+                  lookup_str);
+            return RETURN_ERR;
+        }
+
+        char path[MAX_UCI_BUF_LEN] = "";
+        //get from wireless.radio?.param the param wanted
+        int status = snprintf_s(path, MAX_UCI_BUF_LEN, "wireless.%s.%s", deviceName, param);
+        if (status <= 0) {
+            ERROR("%s failed snprintf. device=%s param=%s\n", __func__, deviceName, param);
+            return RETURN_ERR;
+        }
+
+        status = cfg_uci_get(path, value, MAX_UCI_BUF_LEN);
+        if (status == RETURN_OK) {
+            // DEBUG("%s index=%d %s=%s\n", __func__, index, param, value);
+        } else {
+            ERROR("%s option N/A. device=%s param=%s\n", __func__, deviceName, param);
+        }
+
+        return status;
+    }
+
+    //* type == TYPE_VAP *
+    // read the value from the selected section
+    if (is_section_found) {
+        // get param in the selected section
+        uci_foreach_element(&s->options, n)
+        {
+            struct uci_option *o = uci_to_option(n);
+            // if param is found in options
+            if (strncmp(n->name, param, MAX_UCI_BUF_LEN) == 0 && o->type == UCI_TYPE_STRING) {
+                strncpy_s(value, MAX_UCI_BUF_LEN, o->v.string, MAX_UCI_BUF_LEN - 1);
+                uci_free_context(ctx);
+                return RETURN_OK;
+            }
+        }
+
+        // param not found in option
+        ERROR("%s, interface(%s) found but param(%s) isn't configured\n", __FUNCTION__,
+              interface_name, param);
+        uci_free_context(ctx);
+        return RETURN_ERR;
+    }
+
+    //interface not found in wireless
+    ERROR("%s, interface(%s) not found", __FUNCTION__, interface_name);
+    uci_free_context(ctx);
+    return RETURN_ERR;
```

Comment:
So I think this bit should move up before the type == RADIO condition.

Reply:
done as a response to the previous comment. thanks